x/mongo/driver/compression_test.go

```diff
@@ -0,0 +1,39 @@
+package driver
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/stretchr/testify/require"
+
+	"go.mongodb.org/mongo-driver/x/mongo/driver/wiremessage"
+)
+
+func TestCompression(t *testing.T) {
+	compressors := []wiremessage.CompressorID{
+		wiremessage.CompressorNoOp,
+		wiremessage.CompressorSnappy,
+		wiremessage.CompressorZLib,
+		wiremessage.CompressorZstd,
```

Comment:
Does this test actually run if `cgo` is disabled?

Reply:
It does, but there was both a bug (forgot to implement decompress) and the test fails. I've fixed both